libs/grout/src/util.ts

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable vx/gts-jsdoc */
+
+export function isBoolean(value: unknown): value is boolean {
+  return typeof value === 'boolean';
+}
+
+export function isNumber(value: unknown): value is number {
+  return typeof value === 'number';
+}
+
+export function isString(value: unknown): value is string {
+  return typeof value === 'string';
+}
+
+export function isArray(value: unknown): value is unknown[] {
+  return Array.isArray(value);
+}
+
+export function isObject(value: unknown): value is Record<string, unknown> {
+  return typeof value === 'object' && !isArray(value) && value !== null;
+}
+
+export function isPlainObject(
+  value: unknown
+): value is Record<string, unknown> {
+  return isObject(value) && value.constructor === Object;
```

Comment:
A safer way to do this:
```suggestion
  return isObject(value) && Object.prototype.toString.call(value) === '[object Object]';
```
If we somehow get an object from another sandbox then its `Object` and our `Object` won't match up.

Reply:
Awesome, thanks!

Reply:
Hmm, looks like this approach thinks a `Result` is a plain object.

```
> const { ok } = require('@votingworks/types')
undefined
> Object.prototype.toString.call(ok())
'[object Object]'
```

I think the goal of this function is to differentiate plain objects from instances of classes. Any other ideas?

Reply:
Maybe `value.constructor.name === 'Object'`?

Reply:
After some researching on SO, I came up with this:

```ts
export function isPlainObject(
  value: unknown
): value is Record<string, unknown> {
  return (
    isObject(value) &&
    Object.getPrototypeOf(value) === Object.prototype &&
    value.constructor.name === Object.name
  );
}
```